core/blockchain.go

```diff
@@ -355,6 +356,21 @@ func (bc *BlockChain) GetVMConfig() *vm.Config {
 	return &bc.vmConfig
 }
 
+// NewSystemEVMRunner creates the System's EVMRunner for given header & sttate
+func (bc *BlockChain) NewSystemEVMRunner(header *types.Header, state vm.StateDB) vm.EVMRunner {
+	return vmcontext.NewSystemEVMRunner(bc, header, state)
+}
+
+// NewSystemEVMRunnerForCurrentBlock creates the System's EVMRunner for current block & state
+func (bc *BlockChain) NewSystemEVMRunnerForCurrentBlock() (vm.EVMRunner, error) {
+	header := bc.CurrentHeader()
```

Comment:
Can `CurrentHeader()` ever return `nil`?  I never thought of it until I saw this nil check in `validatorRandomnessAtBlockNumber()` (in code which is changed in this PR): https://github.com/celo-org/celo-blockchain/blob/master/consensus/istanbul/backend/backend.go#L738-L740

Reply:
I believe not, since `NewHeaderChain()` already sets it and use it. And it should be a big bug if someone did `currentHeader.Store(nil)`